server.js

```diff
@@ -3906,6 +3906,71 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// GitHub languages integration.
+camp.route(/^\/github\/languages\/(top|count|bytes)\/([^\/]+)\/([^\/]+)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var type = match[1];
+  var user = match[2];
+  var repo = match[3];
+  var format = match[4];
+  var apiUrl = githubApiUrl + '/repos/' + user + '/' + repo + '/languages';
+  var badgeData = getBadgeData('', data);
+  if (badgeData.template === 'social') {
+    badgeData.logo = getLogo('github', data);
+  }
+  githubAuth.request(request, apiUrl, {}, function(err, res, buffer) {
+    if (err != null) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+      return;
+    }
+    try {
+      const parsedData = JSON.parse(buffer);
+      var sumBytes = 0;
+      switch(type) {
+        case 'top':
+          var topLanguage = 'language';
+          var maxBytes = 0;
+          for (const language of Object.keys(parsedData)) {
+            const bytes = parseInt(parsedData[language]);
+            if (bytes >= maxBytes) {
+              maxBytes = bytes;
+              topLanguage = language;
+            }
+            sumBytes += bytes;
+          }
+          badgeData.text[0] = topLanguage;
+          if (sumBytes === 0) { // eg, empty repo, only .md files, etc.
+            badgeData.text[1] = 'none';
+            badgeData.colorscheme = 'blue';
+          } else {
+            badgeData.text[1] = (maxBytes / sumBytes * 100).toFixed(1) + '%'; // eg, 9.1%
+          }
+          break;
+        case 'count':
+          badgeData.text[0] = 'languages';
+          badgeData.text[1] = Object.keys(parsedData).length;
+          badgeData.colorscheme = 'blue';
+          break;
+        case 'bytes':
+          for (const language of Object.keys(parsedData)) {
+            sumBytes += parseInt(parsedData[language]);
+          }
+          badgeData.text[0] = 'code size';
+          badgeData.text[1] = metric(sumBytes) + 'B';
+          badgeData.colorscheme = 'blue';
```

Comment:
How would you get the code size directly using that endpoint? The `size` parameter seemed to be a good candidate, but must actually relate to something else, as it indicates only 4945 for this repository for instance.

Reply:
I think that is right. It's in kilobytes. Here's a [StackOverflow answer](https://stackoverflow.com/a/8679592/893113) that talks about it.

Reply:
The more I think about it, the repo size is more about how many revisions and binary blobs are checked in. The size of the identifiable code could be useful, distinct from the repo size. 

Do they provide lines of code? That would also be really cool. 😃

Reply:
Yes, I think it's the size on disk rather than the code size. I would be inclined to keep the `code size` badge as it is, but I can also provide a `repo size` badge through a new pull request, if you want!
They do not provide lines of code directly, but this [StackOverflow answer](https://stackoverflow.com/a/32154303/3527464) suggests a nice approach to get it. Yet another pull request coming up? :smile:

Reply:
Awesome! Yes, a separate badge + PR for repo size makes sense to me.

Wow, if that SO answer works, that would be cool!

Reply:
Okay, I'll take of that!

Reply:
Actually the solutions proposed to calculate the lines of code using the API are dodgy according to a [comment](https://stackoverflow.com/questions/26881441/can-you-get-the-number-of-lines-of-code-from-a-github-repository/32154303#comment63416385_32154303) on the answer I had previously linked. I don't think a line count badge can be implemented reliably and efficiently with the current API possibilities.